# Hospital Readmission Risk Analysis

## Exploratory Data Analysis (EDA)

Author: Dimitris Artinopoulos

Purpose:
Initial exploration of the synthetic hospital readmission dataset.

## 1. Import Libraries

In [1]:
import pandas as pd 

## 2. Load Dataset

In [2]:
df = pd.read_csv("../data/diabetic_data.csv")

## 3. Preview the Dataset

In [3]:
df.head()

,encounter_id,patient_nbr,race,gender,age,weight,admission_type_id,discharge_disposition_id,admission_source_id,time_in_hospital,...,citoglipton,insulin,glyburide-metformin,glipizide-metformin,glimepiride-pioglitazone,metformin-rosiglitazone,metformin-pioglitazone,change,diabetesMed,readmitted
0,2278392,8222157,Caucasian,Female,[0-10),?,6,25,1,1,...,No,No,No,No,No,No,No,No,No,NO
1,149190,55629189,Caucasian,Female,[10-20),?,1,1,7,3,...,No,Up,No,No,No,No,No,Ch,Yes,>30
2,64410,86047875,AfricanAmerican,Female,[20-30),?,1,1,7,2,...,No,No,No,No,No,No,No,No,Yes,NO
3,500364,82442376,Caucasian,Male,[30-40),?,1,1,7,2,...,No,Up,No,No,No,No,No,Ch,Yes,NO
4,16680,42519267,Caucasian,Male,[40-50),?,1,1,7,1,...,No,Steady,No,No,No,No,No,Ch,Yes,NO


## 4. Understand the Dataset Structure

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 101766 entries, 0 to 101765
Data columns (total 50 columns):
 #   Column                    Non-Null Count   Dtype
---  ------                    --------------   -----
 0   encounter_id              101766 non-null  int64
 1   patient_nbr               101766 non-null  int64
 2   race                      101766 non-null  str  
 3   gender                    101766 non-null  str  
 4   age                       101766 non-null  str  
 5   weight                    101766 non-null  str  
 6   admission_type_id         101766 non-null  int64
 7   discharge_disposition_id  101766 non-null  int64
 8   admission_source_id       101766 non-null  int64
 9   time_in_hospital          101766 non-null  int64
 10  payer_code                101766 non-null  str  
 11  medical_specialty         101766 non-null  str  
 12  num_lab_procedures        101766 non-null  int64
 13  num_procedures            101766 non-null  int64
 14  num_medications           10176

### Observations

- The dataset contains 101,766 hospital encounters.
- It includes 50 features.
- There are 13 numerical and 37 categorical variables.
- The target variable appears to be 'readmitted'.
- Missing values are not detected automatically because some missing entries are likely stored as "?" strings.

## 5. Data Quality Assessment

In [5]:
df.isnull().sum()

encounter_id                    0
patient_nbr                     0
race                            0
gender                          0
age                             0
weight                          0
admission_type_id               0
discharge_disposition_id        0
admission_source_id             0
time_in_hospital                0
payer_code                      0
medical_specialty               0
num_lab_procedures              0
num_procedures                  0
num_medications                 0
number_outpatient               0
number_emergency                0
number_inpatient                0
diag_1                          0
diag_2                          0
diag_3                          0
number_diagnoses                0
max_glu_serum               96420
A1Cresult                   84748
metformin                       0
repaglinide                     0
nateglinide                     0
chlorpropamide                  0
glimepiride                     0
acetohexamide 

### Observations

- The dataset contains missing values.
- The largest number of missing values appears in 'max_glu_serum' and 'A1Cresult'.
- Most other variables are complete.
- Missing values may reflect that these laboratory tests were not performed for every patient.

In [6]:
missing_percentage = (df.isnull().sum() / len(df)) * 100
missing_percentage.sort_values(ascending=False)

max_glu_serum               94.746772
A1Cresult                   83.277322
race                         0.000000
gender                       0.000000
age                          0.000000
weight                       0.000000
admission_type_id            0.000000
discharge_disposition_id     0.000000
admission_source_id          0.000000
time_in_hospital             0.000000
payer_code                   0.000000
medical_specialty            0.000000
num_lab_procedures           0.000000
num_procedures               0.000000
num_medications              0.000000
number_outpatient            0.000000
encounter_id                 0.000000
patient_nbr                  0.000000
number_inpatient             0.000000
number_emergency             0.000000
diag_1                       0.000000
diag_2                       0.000000
number_diagnoses             0.000000
diag_3                       0.000000
metformin                    0.000000
repaglinide                  0.000000
nateglinide 

### Observations

- 'max_glu_serum' has approximately 94.7% missing values.
- 'A1Cresult' has approximately 83.3% missing values.
- All remaining variables are complete.
- These missing values may indicate that these laboratory tests were not performed for most patients rather than representing data entry errors.

## 7. Explore the Target Variable

In [7]:
df["readmitted"].value_counts()

readmitted
NO     54864
>30    35545
<30    11357
Name: count, dtype: int64

### Observations

- The target variable is `readmitted`.
- The dataset contains three classes:
  - `NO`
  - `>30`
  - `<30`
- Most patients were not readmitted.
- The `<30` category is the smallest and represents patients readmitted within 30 days.
- Predicting early readmission could help hospitals improve patient care and reduce healthcare costs.